#!/usr/bin/env python
PACKAGE = "CORC"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("ff_ratio", double_t, 0, "Feedforward ratio", 0.72, 0, 1)
gen.add("kp", double_t, 0, "P", 1.08, 0, 5)
gen.add("ki", double_t, 0, "I", 0.17, 0, 2)
gen.add("kd", double_t, 0, "D", 0, 0, 0.1)

interaction_enum = gen.enum([
                   gen.const("zero_calibration", int_t, 0, "controller 0"),
                   gen.const("zero_torque", int_t, 1, "controller 1"),
                   gen.const("virtual_spring", int_t, 2, "controller 2")],
                    "MultiState Controller Mode")

gen.add("controller_mode", int_t, 1, "Controller Mode", 1, 0, 3, edit_method=interaction_enum) # 3 is the maximum value

exit(gen.generate(PACKAGE, "CORC", "dynamic_params"))